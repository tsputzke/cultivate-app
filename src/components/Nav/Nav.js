import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import UserContext from '../../context/user-context'

export default class Nav extends Component {
  static contextType = UserContext

  // Clear sessionStorage on logout
  handleLogoutClick = () => {
    sessionStorage.clear()
    this.context.toggleState()
  }

  render() {
    return (
      <nav className='nav'>
        <ul className='nav-links'>
          <li id="nav-button-left"><Link to='/about'>About</Link></li>
          <li><Link to='/show-user'>Home</Link></li>
          <li><Link onClick={this.handleLogoutClick} to='/'>Logout</Link></li>
        </ul>
      </nav>
    )
  }
}
