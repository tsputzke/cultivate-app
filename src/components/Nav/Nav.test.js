import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Nav from './Nav'

configure({adapter: new Adapter()});  

describe(`Nav component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders Nav by default', () => {
    const wrapper = shallow(<Nav />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})