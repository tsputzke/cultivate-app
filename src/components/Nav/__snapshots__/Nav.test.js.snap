// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders Nav by default 1`] = `
<nav
  className="nav"
>
  <section
    className="nav-links"
  >
    <div
      className="nav-button left-nav"
    >
      <Link
        to="/about"
      >
        About.
      </Link>
    </div>
    <div
      className="home-icon right-nav"
    >
      <Link
        to="/show-user"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f015",
                "M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z",
              ],
              "iconName": "home",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </Link>
    </div>
    <div
      className="nav-button right-nav"
    >
      <Link
        onClick={[Function]}
        to="/"
      >
        Logout.
      </Link>
    </div>
  </section>
</nav>
`;
